algorithms task (5)/student/include/impl.h

```diff
@@ -10,4 +13,12 @@
  * @return gathered range
  */
 template <typename BiIt, typename UnaryPredicate>
-std::pair<BiIt, BiIt> gather(BiIt begin, BiIt end, BiIt position, UnaryPredicate predicate);
+std::pair<BiIt, BiIt> gather(BiIt begin, BiIt end, BiIt position, UnaryPredicate predicate)
+{
+    const auto partitioningEnd = std::partition(begin, end, predicate);
+    const auto gatheringOffset = std::min(std::distance(begin, position),
```

Comment:
distance call + rotate is not effective operation here, please consider  simpler algorithm

Reply:
Found and implemented new algorithm.
Please take in account my new unit tests condition about that "right" element must be at specified position.